### DISCUSSION TITLE
Graphics with Arch Linux

### DISCUSSION THREAD

**Discussion by emmyoh** [utmapp/UTM#5375] [General]
Hello,

I wanted to share how I managed to transform the Arch Linux VM from being purely command-line to having a desktop environment.

First, I removed the 'Serial' device, and added a 'Display', using `virtio-gpu-pci`. Next, for working audio, I changed the emulated audio card to `ich9-intel-hda`.

I then logged in as `root`, and ran the following commands:

```
pacman -Syu qemu-guest-agent gnome gparted spice-vdagent vulkan-virtio qemu-hw-display-virtio-gpu-pci qemu-hw-display-virtio-gpu-pci-gl networkmanager pipewire pipewire-alsa pipewire-pulse pipewire-jack wireplumber

systemctl enable --now gdm

systemctl enable --now NetworkManager
```

Modify the first command to use your desktop environment of preference.

I would also recommend increasing the amount of memory given to the VM beyond 2 GiB.

I included the `gparted` package to make it easier to resize your partition & filesystem when you resize the virtual disk.

Hopefully someone finds this useful!

Kind regards,
Emil

**Comment by mister-teddy** [comment]
On Mac M1, I got a black screen issue (like [those issues with Kali Linux](https://github.com/utmapp/UTM/discussions/4539)). I fixed it by installing SDDM and KDE Plasma.
It seems like GDM + Gnome / LightDM + Pantheon (my favourite) is not working with UTM on M1. Hopefully, someone will look at it in the future 🌟

**Comment by rvega** [comment]
Works! Thank you for posting this!
